sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/implementation/OpenAIServerSentEvents.java

```diff
@@ -36,87 +35,65 @@ public final class OpenAIServerSentEvents<T> {
-                        if (split.length != 2) {
-                            String line = reader.readLine();
-                            if (line == null) {
-                                lastLine.set(currentLine);
-                                expectEmptyLine.set(false);
-                                return Flux.fromIterable(values);
-                            }
-                            return Flux.error(new IllegalStateException("Invalid data format " + currentLine));
-                        }
-
-                        String dataValue = split[1];
-                        if (split[1].startsWith(" ")) {
-                            dataValue = split[1].substring(1);
-                        }
+        return mapByteBuffersToEvents();
+    }
 
-                        if (!dataValue.isEmpty() && isValidJson(dataValue)) {
-                            T value = SERIALIZER.readValue(dataValue, type);
-                            values.add(value);
-                            lastLine.set("");
-                        } else {
-                            lastLine.set(currentLine);
-                            expectEmptyLine.set(false);
+    private Flux<T> mapByteBuffersToEvents() {
+        return source
+            .publishOn(Schedulers.boundedElastic())
+            .concatMap(byteBuffer -> {
+                List<T> values = new ArrayList<>();
+                byte[] byteArray = byteBuffer.array();
+                for (byte currentByte : byteArray) {
+                    if (currentByte == 0xA || currentByte == 0xD) {
+                        String currentLine;
+                        try {
+                            currentLine = outStream.toString(StandardCharsets.UTF_8.name());
+                            handleCurrentLine(currentLine, values);
+                        } catch (UnsupportedEncodingException | JsonProcessingException e) {
+                            return Flux.error(e);
                         }
+                        outStream = new ByteArrayOutputStream();
                     } else {
-                        expectEmptyLine.set(false);
+                        outStream.write(currentByte);
                     }
-                    currentLine = reader.readLine();
+                }
+                try {
+                    handleCurrentLine(outStream.toString(StandardCharsets.UTF_8.name()), values);
+                    outStream = new ByteArrayOutputStream();
+                } catch (IllegalStateException | JsonProcessingException e) {
+                    // return the values collected so far, as this could be because the server sent event is
+                    // split across two byte buffers and the last line is incomplete and will be continued in
+                    // the next byte buffer
+                    return Flux.fromIterable(values);
+                } catch (UnsupportedEncodingException e) {
+                    return Flux.error(e);
                 }
```

Comment:
```suggestion
                try {
                    handleCurrentLine(outStream.toString(StandardCharsets.UTF_8), values);
                    outStream = new ByteArrayOutputStream();
                } catch (IllegalStateException | JsonProcessingException e) {
                    // return the values collected so far, as this could be because the server sent event is
                    // split across two byte buffers and the last line is incomplete and will be continued in
                    // the next byte buffer
                    return Flux.fromIterable(values);
                }
```

Reply:
My bad. My changes only apply to java 10+. Since we are targeting java 8. Will close this feedback.